File: lib/android/app/src/main/java/com/reactnativenavigation/views/TopTabs.java
package com.reactnativenavigation.views;

import android.content.Context;
import android.graphics.Typeface;
import android.support.design.widget.TabLayout;

import com.reactnativenavigation.parse.Color;
import com.reactnativenavigation.parse.Number;

public class TopTabs extends TabLayout {
    private final TopTabsStyleHelper styleHelper;

    public TopTabs(Context context) {
        super(context);
        styleHelper = new TopTabsStyleHelper(this);
    }

    public void setFontFamily(int tabIndex, Typeface fontFamily) {
        styleHelper.setFontFamily(tabIndex, fontFamily);
    }

    public int[] getSelectedTabColors() {
        return SELECTED_STATE_SET;
    }

    public int[] getDefaultTabColors() {
        return EMPTY_STATE_SET;
    }

    public void applyTopTabsColors(Color selectedTabColor, Color unselectedTabColor) {
        styleHelper.applyTopTabsColors(selectedTabColor, unselectedTabColor);
    }

    public void applyTopTabsFontSize(Number fontSize) {
        styleHelper.applyTopTabsFontSize(fontSize);
    }
}

File: lib/src/params/options/TopTabs.js
class TopTabs {
  /**
   * @property {string} [selectedTabColor] Selected tab color
   * @property {string} unselectedTabColor Unselected tab color
   * @property {int} fontSize
   */
  constructor(topTabs) {
    this.selectedTabColor = topTabs.selectedTabColor;
    this.unselectedTabColor = topTabs.unselectedTabColor;
    this.fontSize = topTabs.fontSize;
  }
}

module.exports = TopTabs;
